### PR TITLE
Remove STS related functionality

### PR DESCRIPTION
**Pull request by deshankoswatte**
- Migrate STS related functionality from carbon-identity-framework/security-mgt module to wso2-extensions/identity-inbound-auth-sts repo.
- Remove dependencies for wss4j and rampart which in turn avoids packing those jars into the product.
- Added dependency for xmlsec and packed it into the product with the xacml-server-feature(this will export the package org.apache.xml.security instead of wss4j exporting it).
- Updated the WSDLs with the migration changes.

Resolves: https://github.com/wso2/product-is/issues/8350

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java [hunk 1]
```diff
@@ -225,7 +224,7 @@ public static String getPPIDDisplayValue(String value) throws Exception {
         if (log.isDebugEnabled()) {
             log.debug("Generating display value of PPID : " + value);
         }
-        byte[] rawPpid = Base64.decode(value);
```
Comment (madurangasiriwardena) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java:228]: We can use jdk decoder as well?

File: components/security-mgt/org.wso2.carbon.security.mgt.ui/src/main/resources/web/securityconfig/index.jsp [hunk 2]
```diff
@@ -20,8 +20,8 @@
 <%@page import="org.apache.axis2.context.ConfigurationContext" %>
 <%@page
         import="org.wso2.carbon.CarbonConstants" %>
-<%@page import="org.wso2.carbon.security.mgt.stub.config.xsd.SecurityScenarioData" %>
```
Comment (madurangasiriwardena) [inline components/security-mgt/org.wso2.carbon.security.mgt.ui/src/main/resources/web/securityconfig/index.jsp:23]: Not sure if this is the correct approach. This UI component should use the classes from the stub.

File: components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/internal/SecurityMgtServiceComponent.java [hunk 3]
```diff
@@ -62,24 +63,17 @@ public static ConfigurationContext getServerConfigurationContext() {
...
-                    new SecurityConfigAdmin(mainAxisConfig, registryService.getConfigSystemRegistry(), null), null);
-            bundleCtx.registerService(Axis2ConfigurationContextObserver.class.getName(),
-                    new SecurityAxis2ConfigurationContextObserver(), null);
-            bundleCtx.registerService(STSAdminServiceInterface.class, new STSAdminServiceImpl(), null);
             bundleCtx.registerService(KeyStoreManagementService.class.getName(), new KeyStoreManagementServiceImpl(),
                     null);
+            try {
+                addKeystores();
+            } catch (Exception e) {
+                String msg = "Cannot add keystores";
```
Comment (madurangasiriwardena) [inline components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/internal/SecurityMgtServiceComponent.java:72]:
```suggestion
                String msg = "Error while adding key stores.";
```
better to add some more info as well.

Comment (deshankoswatte) [reply components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/internal/SecurityMgtServiceComponent.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/2921/commits/7b2708081125a410e0d660cf4f065fd3131049fd.

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java (hunk 1 above)
Comment (deshankoswatte) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java:228]: Yes. Fixed with https://github.com/wso2/carbon-identity-framework/pull/2921/commits/7b2708081125a410e0d660cf4f065fd3131049fd.

File: components/security-mgt/org.wso2.carbon.security.mgt.ui/src/main/resources/web/securityconfig/index.jsp (hunk 2 above)
Comment (deshankoswatte) [reply components/security-mgt/org.wso2.carbon.security.mgt.ui/src/main/resources/web/securityconfig/index.jsp:23]: It uses the classes from the stub which is represented with `xsd`just need to configure to build it as a stub. But right now seems there are some problem with the generation too will discuss this with you offline.

Comment (deshankoswatte) [reply components/security-mgt/org.wso2.carbon.security.mgt.ui/src/main/resources/web/securityconfig/index.jsp:23]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/2921/commits/c7d87531d9707a98ee5b6f19c9ff1f12770e5cd7.

**Comment by deshankoswatte** [comment]
Re initiated at https://github.com/wso2/carbon-identity-framework/pull/2984.

**Pull request closed without merging**